website/pages/partials/from-1.5/variables/custom-validation.mdx

```diff
@@ -0,0 +1,43 @@
+## Custom Validation Rules
+
+In addition to Type Constraints, you can specify arbitrary custom validation
+rules for a particular variable using one or more `validation` block nested
+within the corresponding `variable` block:
+
+```hcl
+variable "image_id" {
+  type        = string
+  description = "The id of the machine image (AMI) to use for the server."
+
+  validation {
+    condition     = length(var.image_id) > 4 && substr(var.image_id, 0, 4) == "ami-"
+    error_message = "The image_id value must be a valid AMI id, starting with \"ami-\"."
+  }
+}
+```
+
+The `condition` argument is an expression that must use the value of the
+variable to return `true` if the value is valid or `false` if it is invalid.
+The expression can refer only to the variable that the condition applies to,
+and _must not_ produce errors.
+
+If the failure of an expression is the basis of the validation decision, use
+[the `can` function](./functions/can.html) to detect such errors. For example:
+
```

Comment:
This fuction does not exist yet but I will add it in another PR, because it's very powerful here.

Reply:
I think we should not document it until it's merged, just in case this ends up released before you get around to it. Unless it's literally the next thing you are going to work on.

Reply:
This is what I'm going to do right up next, it shouldn't be too hard.

Reply:
AH, I was starting to implement this when I realised reading the code I already _did_ implement this 🤦🏼 . Okay this needs updating.